### PR TITLE
"Clear logs" API endpoint

### PR DESCRIPTION
**Pull request by henrikhodne**
This is made so hub doesn't need access to the logs database anymore.

/cc @svenfuchs

### CODE REVIEW

**Comment by henrikhodne** [comment]
It appears my editor decided to insert some hard tabs in some places. Will fix.

File: lib/travis/logs/app.rb [hunk 1]
```diff
@@ -56,6 +57,21 @@ def initialize(existence = nil, pusher = nil)
       get "/uptime" do
         status 204
       end
+
+      post "/logs/:id/clear" do
```
Comment (meatballhat) [inline lib/travis/logs/app.rb]: curious: why not `DELETE /logs/:id`?

**Comment by meatballhat** [comment]
LGTM :sheep: :it:

File: lib/travis/logs/app.rb (hunk 1 above)
Comment (henrikhodne) [inline lib/travis/logs/app.rb]: My reasoning is that it just blanks out the log, the log record is still in the database. That might just be an implementation detail though?

Comment (meatballhat) [inline lib/travis/logs/app.rb]: Yep, `DELETE` semantics are up to our interpretation.  It's not strictly a resource removal, although if we're allowing access to `/logs/:id` afterward and *not* returning 404, then I'd say going with `POST /logs/:id/clear` is probs better :+1:.

**Pull request merged**